bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/BahmniPatientServiceImpl.java

```diff
@@ -93,4 +99,30 @@ public PatientConfigResponse getConfig() {
         return patientDao.getByAIsToB(aIsToB);
     }
 
+    @Override
+    public List<PatientResponse> luceneHibernateSearch(PatientSearchParameters searchParameters) {
+        List<PatientResponse> luceneHibernateResponse = new ArrayList<>();
+        if(searchParameters.getIdentifier() != null || searchParameters.getName() != null || searchParameters.getCustomAttribute() != null){
```

Comment:
Yes you are right. Strings will be always empty if they contain nothing.

Reply:
If all arguments are empty strings, then we don't search for the patient objects

Reply:
If none of the first 3 parameters are missing and others like program attributes and address are provided, it will get to hibernate search and if nothing is supplied at all, then no search will be called